yt_dlp/extractor/gab.py

```diff
@@ -22,48 +23,46 @@ class GabTVIE(InfoExtractor):
+            'uploader_id': '62454a53299da61b33305901',
+            'thumbnail': 'https://tv.gab.com/image/6254b9f31e03ac67731b035a',
         }
     }]
+    _DOMAIN = 'https://tv.gab.com/'
 
     def _real_extract(self, url):
         id = self._match_id(url).split('-')[-1]
         webpage = self._download_webpage(url, id)
         channel_id = self._search_regex(r'data-channel-id=\"(?P<channel_id>[^\"]+)', webpage, 'channel_id')
-        channel_name = self._search_regex(r'data-channel-name=\"(?P<channel_id>[^\"]+)', webpage, 'channel_name')
-        title = self._search_regex(r'data-episode-title=\"(?P<channel_id>[^\"]+)', webpage, 'title')
-        view_key = self._search_regex(r'data-view-key=\"(?P<channel_id>[^\"]+)', webpage, 'view_key')
-        description = clean_html(
-            self._html_search_regex(self._meta_regex('description'), webpage, 'description', group='content')) or None
-        available_resolutions = re.findall(r'<a\ data-episode-id=\"%s\"\ data-resolution=\"(?P<resolution>[^\"]+)' % id,
-                                           webpage)
-
-        formats = []
-        for resolution in available_resolutions:
-            frmt = {
-                'url': f'https://tv.gab.com/media/{id}?viewKey={view_key}&r={resolution}',
-                'format_id': resolution,
-                'vcodec': 'h264',
-                'acodec': 'aac',
-                'ext': 'mp4'
-            }
-            if 'audio-' in resolution:
-                frmt['abr'] = str_to_int(resolution.replace('audio-', ''))
-                frmt['height'] = 144
-                frmt['quality'] = -10
-            else:
-                frmt['height'] = str_to_int(resolution.replace('p', ''))
-            formats.append(frmt)
+        video_info = self._parse_json(self._search_regex(r'(?s)mediaMetadata:\s+({.+?})', webpage, 'video_info'),
+                                      video_id=id, transform_source=js_to_json)
+        formats = [{
+            'format_id': q,
+            'quality': q,
+            'ext': 'mp4',
+            'url': urllib.parse.urljoin(self._DOMAIN, src),
+        } for src, q in re.findall(r'<source[^>]+src="(?P<src>[^"]+)"[^>]*\ssize="(?P<quality>\d+)"', webpage)]
         self._sort_formats(formats)
 
+        poster = self._search_regex(r'data-poster="(?P<data_poster>[^"]+)', webpage, 'poster', fatal=False)
+
         return {
             'id': id,
-            'title': title,
+            'title': video_info.get('title'),
             'formats': formats,
-            'description': description,
-            'uploader': channel_name,
+            'description': clean_html(self._html_search_regex(self._meta_regex('description'), webpage, 'description', group='content')) or None,
+            'uploader': video_info.get('artist'),
             'uploader_id': channel_id,
-            'thumbnail': f'https://tv.gab.com/image/{id}',
+            'thumbnail': urllib.parse.urljoin(self._DOMAIN, poster) if poster else None,
```

Comment:
```suggestion
            'thumbnail': urljoin('https://tv.gab.com/', poster),
```
Use the urljoin from utils